### PR TITLE
Finalise multiple exits

### PR DESCRIPTION
**Pull request by eezcjkr**
Resolves https://github.com/leapdao/leap-contracts/issues/123

### CODE REVIEW

File: contracts/ExitHandler.sol [hunk 1]
```diff
@@ -180,27 +180,40 @@ contract ExitHandler is DepositHandler {
...
     uint256 exitableAt;
+    Exit memory currentExit;
+
     (utxoId, exitableAt) = getNextExit(_color);
 
-    require(exitableAt <= block.timestamp, "The top exit can not be exited yet");
-    require(tokens[_color].currentSize > 0, "The exit queue for color is empty");
+    for (uint i = 0; i<20; i++) {
+      // if queue is empty or top exit cannot be exited yet, stop
+      if (exitableAt > block.timestamp || tokens[_color].currentSize < 1) {
```
Comment (troggy) [inline contracts/ExitHandler.sol]: > tokens[_color].currentSize < 1

check is unnecessary in the loop. We need to check it once before the loop (to cover "no exits for color" situation, like it was [before](https://github.com/leapdao/leap-contracts/pull/152/files#diff-e8b2bda9737fa1a1ff063bbb22fb4cd5L186)). Later it is covered by condition on [line 211](https://github.com/leapdao/leap-contracts/pull/152/files#diff-e8b2bda9737fa1a1ff063bbb22fb4cd5R211).

File: contracts/ExitHandler.sol [hunk 2]
```diff
@@ -180,27 +180,40 @@ contract ExitHandler is DepositHandler {
   function finalizeTopExit(uint16 _color) public {
```
[the remark this answers is not in the document]
Comment (eezcjkr) [reply contracts/ExitHandler.sol]: This makes sense but will break bridgeUI and integ-tests.

[the remark this answers is not in the document]
Comment (eezcjkr) [reply contracts/ExitHandler.sol]: So not sure if we should go ahead.

[the remark this answers is not in the document]
Comment (johannbarbie) [reply contracts/ExitHandler.sol]: i wouldn't rename, just for the sake of keeping interfaces the same

[the remark this answers is not in the document]
Comment (troggy) [reply contracts/ExitHandler.sol]: alias and deprecation is good idea 👍

[the remark this answers is not in the document]
Comment (eezcjkr) [reply contracts/ExitHandler.sol]: How do you add an alias??

File: contracts/TransferrableToken.sol [hunk 3]
```diff
@@ -12,6 +12,7 @@ import "../node_modules/openzeppelin-solidity/contracts/introspection/ERC165.sol
 
 
 contract TransferrableToken is ERC165 {
+  function transfer(address _to, uint256 _valueOrTokenId) public;
```
Comment (troggy) [inline contracts/TransferrableToken.sol]: sorry, I messed up things here :( There is no `transfer` function in ERC721 interface, that's why we used `transferFrom` for everything.

So probably better remove `transfer` from `TransferrableToken` and either:
1. use approve + transferFrom as it was before
2. cast to ERC20 here https://github.com/leapdao/leap-contracts/pull/152/commits/ab2234c2258a435b5508c2e8d7a330bfefa2c005#diff-e8b2bda9737fa1a1ff063bbb22fb4cd5R202 so `transfer` can be used

**Pull request merged**